common/testing/nettest/rpc_factory_test.go

```diff
@@ -84,3 +264,22 @@ func testDialer(t *testing.T, target string, dial func(rpcFactory *nettest.RPCFa
 func newRPCFactory(dialOptions ...grpc.DialOption) *nettest.RPCFactory {
 	return nettest.NewRPCFactory(nettest.NewListener(nettest.NewPipe()), dialOptions...)
 }
+
+// mockResolver is a custom resolver for DNS URLs
+type mockResolver struct{}
+
+func (r *mockResolver) Build(target resolver.Target, cc resolver.ClientConn, opts resolver.BuildOptions) (resolver.Resolver, error) {
```

Comment:
Given that [operator_handler.go](https://github.com/temporalio/temporal/blob/main/service/frontend/operator_handler.go) does not implement a resolver, I believe tests using this `mockResolver` could yield a false positive.

Reply:
@SakshiMehta in case the operator upsert command is not behaving in line with these unit tests, we should investigate further the operator_handler.go implementation and whether a separate resolver is needed.

Reply:
WDYM `operation_handler.go` does not implement a resolver?